src/concrete_intersection.jl

```diff
@@ -400,7 +400,11 @@ function intersection(P1::HPoly{N},
             backend = GLPKSolverLP()
         end
         # here, detection of empty intersection may be reported as an infeasible LP
-        remove_redundant_constraints!(Q, backend=backend)
+        if remove_redundant_constraints!(Q, backend=backend)
+            return Q
+        else
+            return EmptySet{N}()
+        end
     end
     return Q
```

Comment:
The `return Q` at the end of the function could be pulled inside the `if` case above to make the flow clearer.